core/src/main/scala/kafka/utils/Log4jController.scala

```diff
@@ -22,44 +22,78 @@ import java.util.Locale
 
 import org.apache.log4j.{Level, LogManager, Logger}
 
+import scala.collection.mutable
+import scala.collection.JavaConverters._
 
-/**
- * An MBean that allows the user to dynamically alter log4j levels at runtime.
- * The companion object contains the singleton instance of this class and
- * registers the MBean. The [[kafka.utils.Logging]] trait forces initialization
- * of the companion object.
- */
-private class Log4jController extends Log4jControllerMBean {
 
-  def getLoggers = {
-    val lst = new util.ArrayList[String]()
-    lst.add("root=" + existingLogger("root").getLevel.toString)
+object Log4jController {
+  val ROOT_LOGGER = "root"
+
+  /**
+    * Returns a map of the log4j loggers and their assigned log level.
+    * If a logger does not have a log level assigned, we return the string "null" for that logger
+    */
+  def loggers: mutable.Map[String, String] = {
+    val logs = new mutable.HashMap[String, String]()
+    logs.put(ROOT_LOGGER, existingLogger(ROOT_LOGGER).getLevel.toString)
+
     val loggers = LogManager.getCurrentLoggers
     while (loggers.hasMoreElements) {
       val logger = loggers.nextElement().asInstanceOf[Logger]
       if (logger != null) {
-        val level =  if (logger != null) logger.getLevel else null
-        lst.add("%s=%s".format(logger.getName, if (level != null) level.toString else "null"))
+        val level = if (logger.getLevel != null) logger.getLevel.toString else "null"
```

Comment:
I'm actually not sure if we should return root logger here... it could be confusing. On the other hand, null looks like an error. Maybe something like "not set"?

Reply:
I had left it as `null` since that is what our current Log4jController did and to keep the scope of interface changes low. It would be inconsistent if the MBean returned "null" and the API "not set"

Let's decide on this.
* "not set" - it's just a matter of naming and I agree this is more human readable. I agree we shouldn't leave it as "null" in any case
* root logger - I prefer this. I don't think this will be confusing. If anything, it will be more informative. In practice, if not set explicitly it should always reflect the root logger (unless some crazy hierarchy was configured in log4.properties I guess). As a user, if I see `null` or `not set` I would need to **know** and fetch the root logger.
I think that this is a better approach and I also think that we should apply this to the Log4j MBean as well.
I can update the KIP, mention in the vote thread and we can be good to go.
LMK what you think

Reply:
Yeah, you are right - root logger sounds good.